sdks/java/core/src/main/java/org/apache/beam/sdk/coders/CoderFactories.java

```diff
@@ -69,30 +59,33 @@ private CoderFactories() { } // Static utility class
    * {@code fromStaticMethods(ListCoder.class)}
    * will produce a {@code Coder<List<X>>} for any {@code Coder Coder<X>}.
    */
-  public static <T> CoderFactory fromStaticMethods(Class<T> clazz) {
-    return new CoderFactoryFromStaticMethods(clazz);
+  public static <T> CoderFactory fromStaticMethods(Class<T> rawType, Class<?> coderClazz) {
+    return new CoderFactoryFromStaticMethods(rawType, coderClazz);
   }
 
   /**
    * Creates a {@link CoderFactory} that always returns the
    * given coder.
-   *
-   * <p>The {@code getInstanceComponents} method of this
-   * {@link CoderFactory} always returns an empty list.
    */
-  public static <T> CoderFactory forCoder(Coder<T> coder) {
-    return new CoderFactoryForCoder<>(coder);
+  public static <T> CoderFactory forCoder(TypeDescriptor<T> type, Coder<T> coder) {
+    return new CoderFactoryForCoder<>(type, coder);
   }
 
   /**
    * See {@link #fromStaticMethods} for a detailed description
    * of the characteristics of this {@link CoderFactory}.
    */
-  private static class CoderFactoryFromStaticMethods implements CoderFactory {
+  private static class CoderFactoryFromStaticMethods<T> implements CoderFactory {
```

Comment:
This shouldn't be parameterized. The `<T>` comes as a generic param to the method below.